src/MongoDB.Driver/Linq/Linq3Implementation/Ast/Optimizers/AstPipelineOptimizer.cs

```diff
@@ -13,14 +13,15 @@
 * limitations under the License.
 */
 
-namespace MongoDB.Driver.Linq.Linq3Implementation.Ast.PipelineOptimizer
+namespace MongoDB.Driver.Linq.Linq3Implementation.Ast.Optimizers
 {
     internal static class AstPipelineOptimizer
     {
         public static AstPipeline Optimize(AstPipeline pipeline)
         {
-            var groupOptimizer = new AstGroupPipelineOptimizer();
-            return groupOptimizer.Optimize(pipeline);
+            pipeline = AstGroupPipelineOptimizer.Optimize(pipeline);
+            pipeline = AstSimplifier.SimplifyAndConvert(pipeline);
```

Comment:
Rename `SimplifyAndConvert` to `Optimize` for consistency with `AstGroupPipelineOptimizer`.

Reply:
`Optimize` isn't as good as `Simplify` because it only simplifies, it doesn't optimize.

My inclination is to leave the name as is.